tests/test_production_flows.py

```diff
@@ -51,6 +51,7 @@ def test_p1_p2_flow_should_create_correct_structure():
 
     for _, data in G.nodes(data=True):
         assert data in expected_nodes
+        expected_nodes.remove(data)
```

Comment:
why dont you just do something like this?
`    assert [(1, {'label': 'el', 'pos': (0.5, 0.5), 'layer': 0}),
            (2, {'label': 'i', 'pos': (0.5, 0.5), 'layer': 1}),
            (3, {'label': 'E', 'pos': (0, 0), 'layer': 1}),
            (4, {'label': 'E', 'pos': (1, 0), 'layer': 1}),
            (5, {'label': 'E', 'pos': (0, 1), 'layer': 1}),
            (6, {'label': 'E', 'pos': (1, 1), 'layer': 1}),
            (7, {'label': 'I', 'pos': (0.25, 0.5), 'layer': 2}),
            (8, {'label': 'I', 'pos': (0.75, 0.5), 'layer': 2}),
            (9, {'label': 'E', 'pos': (0, 0), 'layer': 2}),
            (10, {'label': 'E', 'pos': (0.5, 0.0), 'layer': 2}),
            (11, {'label': 'E', 'pos': (1, 0), 'layer': 2}),
            (12, {'label': 'E', 'pos': (0, 1), 'layer': 2}),
            (13, {'label': 'E', 'pos': (0.5, 1.0), 'layer': 2}),
            (14, {'label': 'E', 'pos': (1, 1), 'layer': 2})] == list(G.nodes(data=True))`

Reply:
Because it requires me to map out the exact id to the data which is assigned arbitrarily and has nothing to do with the correct structure. Also if we change the order in which we add the nodes your test will fail, when it should pass.